core/src/main/java/org/mule/runtime/core/internal/streaming/bytes/AbstractCursorStream.java

```diff
@@ -72,7 +78,7 @@ public CursorStreamProvider getProvider() {
 
   protected void assertNotDisposed() throws IOException {
     if (released) {
-      throw new IOException("Stream is closed");
+      throw new IOException("Stream is closed", closerResponsible);
```

Comment:
if `ClosingCursorException ` extended `IOException you could throw it directly`, avoiding the cost of creating a new exception